bindings/node/index.js

```diff
@@ -15,27 +23,20 @@ function extension(mongodb) {
     MongoCryptError
   };
 }
-let _module;
-function loadModule() {
-  if (!_module) {
-    _module = extension(require('mongodb'));
-  }
 
-  return _module;
-}
-
-function memo(fn) {
-  let value;
-  return () => value || (value = fn());
-}
-
-exports.extension = extension;
-exports.MongoCryptError = MongoCryptError;
-Object.defineProperty(exports, 'AutoEncrypter', {
-  enumerable: true,
-  get: memo(() => loadModule().AutoEncrypter)
-});
-Object.defineProperty(exports, 'ClientEncryption', {
-  enumerable: true,
-  get: memo(() => loadModule().ClientEncryption)
-});
+module.exports = {
+  extension,
+  MongoCryptError,
+  get AutoEncrypter() {
```

Comment:
cool, didn't know this worked outside of a `class` definition. This seems relatively complicated with all the deletion, why not just have the getter returned a cached copy? It seems like you already cache it, so can't the getter just be:
```
const m = loadDefaultModule();
return m.AutoEncrypter;
```

Reply:
Eh, I wanted to get rid of the getter. Did a benchmark and it is *technically* better to reset it to a prop:

```js

{ prop:
   { moe: 1.0856058264241745e-10,
     rme: 1.589729254384141,
     sem: 5.5388052368580335e-11,
     deviation: 5.2252984098655e-10,
     mean: 6.828872422334184e-9,
     variance: 2.730374347214292e-19 },
  getter:
   { moe: 9.04029618171838e-11,
     rme: 0.87392302730372,
     sem: 4.6123960110808064e-11,
     deviation: 4.448033630226029e-10,
     mean: 1.034449934293418e-8,
     variance: 1.9785003175621743e-19 } }
```

but honestly thats a micro-optimization. We can just leave them as getters if you want.

Reply:
the getter feels less complicated to me, my priority is readability for the future so I'd prefer to just leave them as getters